### PR TITLE
Iqbal

### PR DESCRIPTION
**Pull request by iqbalskd**
Please review Issue#7

I will update the Read Me file as well if this is good to go.

### CODE REVIEW

File: Model/RecoveryItem.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@ namespace CustomWatchdog
     internal class RecoveryItem
     {
         public string RecoveryBatch { get; set; }
+        public uint overrideRecoveryExecutionTimeout { get; set; }
```
Comment (ubbeK) [inline Model/RecoveryItem.cs]: I know that this repo might need some cleaning up, but lets try to keep to our coding standards, https://github.com/Starcounter/CompanyTrack/blob/15235f13eee21b97625fe6845063d92d68d63516/AppsTeam/Guidelines/coding_style/csharp_code_formatting.md

This should be called `OverrideRecoveryExecutionTimeout`

File: CustomBatchWatchdog.cs [hunk 2]
```diff
@@ -117,6 +123,8 @@ private void LoadConfigFromFile()
         private void Recover(RecoveryItem rc)
         {
             ApplicationLoader.PROCESS_INFORMATION procInfo;
+            if (rc.overrideRecoveryExecutionTimeout != 0)
+                recoveryExecutionTimeout = rc.overrideRecoveryExecutionTimeout;
```
Comment (ubbeK) [inline CustomBatchWatchdog.cs]: Please use the [Allman style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style) when it comes to braces, you even did it between rows 76-79 in this file.

```csharp
if (rc.OverrideRecoveryExecutionTimeout != 0)
{
    recoveryExecutionTimeout = rc.OverrideRecoveryExecutionTimeout;
}
```

File: CustomBatchWatchdog.cs [hunk 3]
```diff
@@ -21,8 +21,8 @@ public partial class CustomBatchWatchdog : ServiceBase
         List<RecoveryItem> recoveryItems = new List<RecoveryItem>();
 
         // local constants (not configurable)
```
Comment (ubbeK) [inline CustomBatchWatchdog.cs]: remove this comment, they are now configurable

File: CustomBatchWatchdog.cs [hunk 4]
```diff
@@ -21,8 +21,8 @@ public partial class CustomBatchWatchdog : ServiceBase
...
-        const string configFileName = "cbwatchdog.json";
-        const string eventLogSource = "Custom Batch Watchdog";
+        string configFileName = "cbwatchdog.json";
+        string eventLogSource = "Custom Batch Watchdog";
```
Comment (ubbeK) [inline CustomBatchWatchdog.cs]: please specify the default visibility (`private`) to all the fields above.

https://github.com/Starcounter/CompanyTrack/blob/15235f13eee21b97625fe6845063d92d68d63516/AppsTeam/Guidelines/coding_style/csharp_code_formatting.md -> 7

File: CustomBatchWatchdog.cs [hunk 5]
```diff
@@ -207,13 +218,36 @@ private void RunForever()
...
             LoadConfigFromFile();
             ThreadPool.QueueUserWorkItem(o => { RunForever(); });
         }
+        private void OverrideSettings(string[] args)
+        {
+            if (args.Length > 0)
+            {
+                configFileName = args[0];
+                if (configFileName.IndexOf('/') > -1)
+                    configFileName = configFileName.Replace("/", string.Empty);
```
Comment (ubbeK) [inline CustomBatchWatchdog.cs]: Use the [Allman style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style) for braces

```csharp
if (configFileName.IndexOf('/') > -1)
{
    configFileName = configFileName.Replace("/", string.Empty);
}
```

We usually use `-` (short version) and `--` (long version) for CLI options, see `star.exe`:

```
C:\Github\Starcounter\cbwatchdog [Iqbal ≡]> star.exe
Usage: star [options] executable [parameters]

Options:
  -h, --help            Shows help about star.exe.
  -hx, --helpextended   Shows extended help about star.exe.
  -v, --version         Prints the version of Starcounter.
  -i, --info            Prints information about the Starcounter installation.
  -p=,--serverport=port Port to the admin server.
  --server=name         Specifies the name of the server. If no port is
                        specified, star.exe use the known port of server.
  --serverhost=host     Specifies the identity of the server host. If no
                        host is specified, 'localhost' is used.
  -d=,--database=name   The database to use. 'Default' is used if not given.
                        Example "star d=foo bar.exe"
  --stop                Stops the given application.
  --norestart           Prevent the application from being restarted if running.
  --name=name           Gives the application the specified name.
  -t, --transact        Runs the entrypoint in the scope of a write transaction
```

or `stardump.exe unload -?`

```
C:\Github\Starcounter\cbwatchdog [Iqbal ≡]> stardump unload -?


Usage:  unload [options]

Options:
  -db | --database <DatabaseName>               Name of starcounter database to dump. Default value is "default"
  -f | --file <FileName>                        Output dump filename
  -b | --buffersize <BufferSize>                Set insert rows buffer size to dump database.
  -scp | --skipcolumnprefixes <ColumnPrefixes>  Column prefixes to skip, space and/or comma separated. Example: -scp="a b,c". Default value is "__"
  -stp | --skiptableprefixes <TablePrefixes>    Table prefixes to skip, space and/or comma separated. Example: -stp="a b,c"
  -st | --skiptables <TableNames>               Table names to skip, space and/or comma separated. Example: -stp="a b,c"
  -ut | --unloadtables <TableNames>             Table names to unlaod, space and/or comma separated. Example: -stp="a b,c"
  -V | --verbose <Verbose>                      Verbose mode, higher number prints more. Default number is 2 [Error:0, Warning:1, Notice:2, Info:3, Debug:4]
  -? | -h | --help                              Show help information
```

File: ProjectInstaller.Designer.cs [hunk 6]
```diff
@@ -23,26 +23,26 @@ partial class ProjectInstaller {
...
+            // CustomBatchWatchdog
+            // 
+            this.CustomBatchWatchdog.Description = "Watches a set of given apps and runs the batch file if one is suddenly missing";
+            this.CustomBatchWatchdog.DisplayName = "Custom Batch Watchdog";
+            this.CustomBatchWatchdog.ServiceName = "CustomBatchWatchdog";
+            this.CustomBatchWatchdog.AfterInstall += new System.Configuration.Install.InstallEventHandler(this.serviceInstaller1_AfterInstall);
+            // 
+            // ProjectInstaller
+            // 
+            this.Installers.AddRange(new System.Configuration.Install.Installer[] {
```
Comment (ubbeK) [inline ProjectInstaller.Designer.cs:50]: please fix the indent here, should be 4 spaces

**Comment by ubbeK** [comment]
Update [README.md](https://github.com/Starcounter/cbwatchdog/tree/Iqbal) with
- how the Service parameters may be used
    - What happens if configure file name is overridden
- `overrideRecoveryExecutionTimeout` in json example and what it does

Update https://github.com/Starcounter/cbwatchdog/blob/Iqbal/cbwatchdog.json example with `overrideRecoveryExecutionTimeout`

**Comment by ubbeK** [comment]
> Please review Issue#7
>
> I will update the Read Me file as well if this is good to go.

Sorry, read this now

File: CustomBatchWatchdog.cs (hunk 3 above)
Comment (iqbalskd) [reply CustomBatchWatchdog.cs]: Done

File: CustomBatchWatchdog.cs (hunk 4 above)
Comment (iqbalskd) [reply CustomBatchWatchdog.cs]: Done

File: CustomBatchWatchdog.cs (hunk 2 above)
Comment (iqbalskd) [reply CustomBatchWatchdog.cs]: Done

File: CustomBatchWatchdog.cs (hunk 5 above)
Comment (iqbalskd) [reply CustomBatchWatchdog.cs]: The net start command doesn't accept both - and --. It only accepts /

File: Model/RecoveryItem.cs (hunk 1 above)
Comment (iqbalskd) [reply Model/RecoveryItem.cs]: Done

File: ProjectInstaller.Designer.cs (hunk 6 above)
Comment (iqbalskd) [reply ProjectInstaller.Designer.cs:50]: Done

File: README.md [hunk 7]
```diff
@@ -41,6 +45,7 @@ The full structure of ```cbwatchdog.json``` is as following:
     {
       "recoveryBatch": "cbwatchdog.bat",
       "scDatabase": "default",
+      "overrideRecoveryExecutionTimeout":"10000",
```
Comment (ubbeK) [inline README.md]: Please add a space after semi column like the rest of the literals. If we don't fix it now, then it most likely will never be fixed.

Also, add this line to one of the items in the example JSON file, https://github.com/Starcounter/cbwatchdog/blob/Iqbal/cbwatchdog.json

**Review by ubbeK — APPROVED** [review]

**Comment by ubbeK** [comment]
Please merge

**Pull request merged**